pkg/config/log.go

```diff
@@ -105,19 +112,34 @@ func SetupLogger(loggerName LoggerName, logLevel, logFile, syslogURI string, sys
 		seelogConfig.ConfigureSyslog(syslogURI, useTLS)
 	}
 
-	configTemplate, err := seelogConfig.Render()
-	if err != nil {
+	if loggerName == "JMX" {
+		jmxSeelogConfig := seelogConfig
```

Comment:
seelogConfig (and jmxSeelogConfig) type is `*seelogCfg.Config`, so subsequent change to `jmxSeelogConfig` will be reflected in `seelogConfig` because they are and remain the same. This break the log level runtime change feature: it reuses the existing `seelogConfig` to just update the log level, but with this change several values are overridden when the `setupLogger` is called for the JMX logger and thus changing the log level actually results in the agent having its main logger reconfigured with settings intended for jmxfetch logging.  You should create a new seelogConfig object to store jmx logging settings.